src/celeritas/ext/ScopedGeantExceptionHandler.hh

```diff
@@ -0,0 +1,51 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/ext/ScopedGeantExceptionHandler.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include <memory>
+
+#include "celeritas_config.h"
+
+class G4VExceptionHandler;
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Install and clear a Geant exception handler during this class lifetime.
+ *
+ * Note that creating a \c G4RunManagerKernel resets the exception
+ * handler, so errors thrown during setup *CANNOT* be caught by Celeritas, and
+ * this class can only be used after creating the \c G4RunManager.
+ */
+class ScopedGeantExceptionHandler
```

Comment:
I recommend that the commit message includes mentioned that this class moved from detail to API and some reasoning for the move.  (This may or may not be independent from the change in name and semantic)